frontend/src/Editor/Viewer/DesktopHeader.jsx

```diff
@@ -19,6 +20,14 @@ const DesktopHeader = ({ showHeader, appName, changeDarkMode, darkMode, setAppDe
     }),
     shallow
   );
+  const { appMode } = useEditorStore(
+    (state) => ({
+      appMode: state.appMode,
+    }),
+    shallow
+  );
+
+  const showDarkModeToggle = useMemo(() => appMode === 'auto', [appMode]);
```

Comment:
Just checking, can we move this logic to the above import? Like as follows,

`  const { appMode, showDarkModeToggle } = useEditorStore(
    (state) => ({
      appMode: state.appMode,
      showDarkModeToggle: appMode === 'auto'
    }),
    shallow
  );`

This way we can avoid `useMemo`. Please check the feasibility on this